core/src/main/resources/application.conf

```diff
@@ -26,12 +26,30 @@ akka {
   }
 
   dispatchers {
-    # A dispatcher for actors with slow/blocking behavior
-    slow-actor-dispatcher {
+    # A dispatcher for actors performing blocking io operations
+    # Prevents the whole system from being slowed down when waiting for responses from external resources for instance
+    io-dispatcher {
       type = Dispatcher
       executor = "fork-join-executor"
       # Using the forkjoin defaults, this can be tuned if we wish
     }
+
+    # A dispatcher for actors handling API operations
+    # Keeps the API responsive regardless of the load of workflows being run
+    api-dispatcher {
+      type = Dispatcher
+      executor = "fork-join-executor"
+    }
+
+    # A dispatcher for engine actors
+    # Because backends behaviour is unpredictable (potentially blocking, slow) the engine runs
+    # on its own dispatcher to prevent backends from affecting its performance.
+    engine-dispatcher {
+      type = Dispatcher
+      executor = "fork-join-executor"
+    }
+
+    # Note that without further configuration, backend actors run on the default dispatcher
```

Comment:
We talked about making something for our own backends, were you going to do that as part of this PR or leave it for a future person?